core/include/traccc/clusterization/detail/measurement_creation_helper.hpp

```diff
@@ -115,6 +116,9 @@ TRACCC_HOST inline void fill_measurement(
                                 pitch[1] * pitch[1] / static_cast<scalar>(12.)};
         // @todo add variance estimation
 
+        // For the ambiguity resolution algorithm, give a unique measurement ID
+        m.measurement_id = unique_mid;
```

Comment:
Instead of any other change, just make this line into:

```suggestion
        m.measurement_id = measurements.size();
```

? This code is currently not shared with the GPU implementation at all (something that I'll want to improve on a little later on), so there's no real need to get the unique ID from the outside into this function. :thinking:

Reply:
You are absolutely right, that's better this way, and only requires to change a single line! :)